holidays/countries/thailand.py

```diff
@@ -302,7 +306,7 @@ def _add_with_observed(dt, holiday_name) -> None:
         #   - CASE 1: SAT-SUN -> 1 in-lieu on TUE
         #   - CASE 2: SUN-MON -> 1 in-lieu on TUE
         # See in lieu logic in `_add_with_observed(dt, holiday_name)`
-        new_years_eve_in_lieu = "New Year's Eve (in lieu)"
+        new_years_eve_in_lieu = self.tr("ชดเชย%s") % self.tr("วันสิ้นปี")
```

Comment:
I believe this is both more readable/optimal:
```suggestion
        new_years_eve_in_lieu = self.tr("ชดเชยวันสิ้นปี")
```

Reply:
I would beg to differ on this one since we've more or less split off all instances of ` (in lieu)` suffixes into its own thing - the only exception being `Special In Lieu Holiday (วันหยุดชดเชย)` as this doesn't fall in the same rule as others.

Doing otherwise would likely require us to add special descriptions for these consecutive holiday in-lieus too (the ones not covered by `_add_with_observed`, which are:
- New Year's Day-New Year's Day (in-lieu)-New Year's Eve (in lieu) combo ← the one mentioned here
- Songkran Festival holidays in Apr
- Asarnha Bucha-Buddhist Lent Day combo in Jul/Aug

Let me know which way you prefer though so that I can have the same loc treatment for all of them @arkid15r

Reply:
That's fine, we can revisit this later. Let's keep it as is for now.